app/scripts/controllers/rif/rns/register/index.js

```diff
@@ -216,4 +218,31 @@ export default class RnsRegister extends RnsJsDelegate {
     return Promise.resolve(transactionListener.id);
   }
 
+  deleteSubdomain (domainName, subdomain) {
+    // TODO: implement using call contract directly because rnsjs doesn't work for send operations.
+    domainName = this.addRskSuffix(domainName);
+    let subdomains = this.getSubdomains(domainName);
+    if (subdomains.filter(sd => sd.name === subdomain).length > 0) {
+      return new Promise((resolve, reject) => {
+        this.rnsJs.subdomains.setOwner(domainName, subdomain, rns.defaultAddress)
```

Comment:
This must be implemented with our custom listener right? 

Also, without using rnsjs for web3 complience issues

Reply:
Yup we need to use our send method and manage the result using the listener.